ui/src/features/BottomPanel/index.tsx

```diff
@@ -24,21 +34,89 @@ const BottomPanel: React.FC<BottomPanelProps> = ({ className }) => {
       description: t("Preview data"),
       component: (
-        <div className="flex flex-1 h-[400px]">
+        <div className={`flex flex-1`}>
           <DataTable />
           <Map />
         </div>
       ),
     },
   ];
 
+  const [selected, setSelected] = useState<PanelContent | undefined>(panelContents?.[0]);
+
+  const handleSelection = useCallback(
+    (content: PanelContent) => {
+      if (content.id !== selected?.id) {
+        setSelected(content);
+        if (!isPanelOpen) {
+          handlePanelToggle?.(true);
+        }
+      } else {
+        handlePanelToggle?.(!isPanelOpen);
+        if (content.id === selected?.id) {
+          setSelected(undefined);
+        }
+      }
+    },
+    [isPanelOpen, handlePanelToggle, selected],
+  );
+
   return (
-    <HorizontalPanel
-      className={`bg-zinc-900 border-t border-zinc-700 backdrop-blur-md ${className}`}
-      isOpen={!!isPanelOpen}
-      panelContents={panelContents}
-      onToggle={handlePanelToggle}
-    />
+    <div
+      // className={`flex flex-col box-content transition-width duration-300 ease-in-out bg-zinc-900 border-t border-zinc-700 backdrop-blur-md ${isPanelOpen ? (windowSize === "max" ? "h-full" : "h-[400px]") : "h-[36px]"}`}>
+      className={`flex flex-col box-content transition-width duration-300 ease-in-out bg-zinc-900 border-t border-zinc-700 backdrop-blur-md`}
+      style={{
+        height: isPanelOpen ? (windowSize === "max" ? "100vh" : "400px") : "36px",
+      }}>
+      <div id="edge" className="flex gap-1 items-center h-[36px]">
+        <div className="flex gap-1 items-center justify-center flex-1 h-[100%]">
+          {panelContents?.map(content => (
+            <IconButton
+              key={content.id}
+              className={`w-[55px] h-[80%] ${selected?.id === content.id ? "text-white bg-zinc-800" : undefined}`}
+              icon={content.icon}
+              tooltipText={content.description}
+              tooltipPosition="top"
+              onClick={() => handleSelection(content)}
+            />
+          ))}
+        </div>
+        {windowSize === "min" && (
+          <IconButton
+            className={`w-[55px] h-[80%]`}
+            icon={<EnterFullScreenIcon />}
+            tooltipText={"Enter full screen"}
+            tooltipPosition="top"
+            onClick={() => setWindowSize("max")}
+          />
+        )}
+        {windowSize === "max" && (
+          <IconButton
+            className={`w-[55px] h-[80%]`}
+            icon={<ExitFullScreenIcon />}
+            tooltipText={"Enter full screen"}
+            tooltipPosition="top"
+            onClick={() => setWindowSize("min")}
+          />
+        )}
+      </div>
+      <div
+        id="content"
+        className={`flex flex-1 bg-zinc-800}`}
+        style={{
+          height: isPanelOpen
+            ? windowSize === "max"
+              ? "calc(100vh - 36px)"
+              : "calc(400px - 36px)"
+            : "0",
+        }}>
```

Comment:
I think this is possible as a tailwind class so please when possible keep styles together 🙏🏻 
Please apply this to the wrapper above too where you split them.
Also, please keep className's consistent so they are simple strings unless passing a variable (className="alsdkfsjdf", not className={`alskdfjkasjdf`})